ci_framework/roles/hci_prepare/tasks/phase2.yml

```diff
@@ -60,6 +60,23 @@
         output_dir: "{{ cifmw_hci_prepare_basedir }}/artifacts"
         script: "oc apply -f {{ cifmw_hci_prepare_basedir }}/artifacts/dpservice-nova-custom-ceph.yml"
 
+    - name: Create Nova SSH Migration Key
+      ci_script:
+        output_dir: "{{ cifmw_hci_prepare_basedir }}/artifacts"
+        script: >-
+          function create_migration_key {
+            pushd "$(mktemp -d)"
+            ssh-keygen -f ./id -t ed25519 -N ''
+            oc create secret generic nova-migration-ssh-key \
+              -n openstack \
```

Comment:
Let's just see if it was variable interpolation that caused this error with patch set 2.

https://review.rdoproject.org/zuul/stream/560fbf4ff40f4507b0ae664a41c63fda?logfile=console.log

```
2023-11-06 15:45:37.711314 | controller | TASK [hci_prepare : Create Nova SSH Migration Key output_dir={{ cifmw_hci_prepare_basedir }}/artifacts, script=function create_migration_key {
2023-11-06 15:45:37.711318 | controller |   pushd "$(mktemp -d)"
2023-11-06 15:45:37.711322 | controller |   ssh-keygen -f ./id -t ed25519 -N ''
2023-11-06 15:45:37.711326 | controller |   oc create secret generic nova-migration-ssh-key \
2023-11-06 15:45:37.711329 | controller |     -n {{ cifmw_install_yamls_defaults.NAMESPACE | default('openstack') }} \
2023-11-06 15:45:37.711334 | controller |     --from-file=ssh-privatekey=id \
2023-11-06 15:45:37.711337 | controller |     --from-file=ssh-publickey=id.pub \
2023-11-06 15:45:37.711341 | controller |     --type kubernetes.io/ssh-auth
2023-11-06 15:45:37.711345 | controller |   rm id*
2023-11-06 15:45:37.711349 | controller |   popd
2023-11-06 15:45:37.711354 | controller | } oc get secret nova-migration-ssh-key {{ cifmw_install_yamls_defaults.NAMESPACE | default('openstack') }} || create_migration_key] ***
2023-11-06 15:45:37.711358 | controller | Monday 06 November 2023  15:45:37 -0500 (0:00:00.600)       0:29:03.329 *******
2023-11-06 15:45:37.845153 | controller | An exception occurred during task execution. To see the full traceback, use -vvv. The error was: NoneType: None
2023-11-06 15:45:37.845192 | controller | fatal: [localhost]: FAILED! => {"changed": true, "msg": "non-zero return code", "rc": 2, "stderr": "", "stderr_lines": [], "stdout": "/home/zuul/.ansible/tmp/ansible-tmp-1699303537.773227-35949-166957789095522/19500c53e296408ab0adc06204133775: line 16: syntax error near unexpected token `oc'\n", "stdout_lines": ["/home/zuul/.ansible/tmp/ansible-tmp-1699303537.773227-35949-166957789095522/19500c53e296408ab0adc06204133775: line 16: syntax error near unexpected token `oc'"]}
```